### PR TITLE
Add recommended k8s labels

### PR DESCRIPTION
**Pull request by eddycharly**
This PR adds recommended k8s labels on resources.
Implementation for proposal #2497 

Config: added labels and updated selectors for all resources in `/config`
Doc: added `/docs/developers/resources-labelling.md`

**NOTE**: this will introduce a breaking change when trying to update deployments as label selectors have changed.

### CODE REVIEW

File: config/101-podsecuritypolicy.yaml [hunk 1]
```diff
@@ -16,6 +16,11 @@ apiVersion: policy/v1beta1
 kind: PodSecurityPolicy
 metadata:
   name: tekton-pipelines
+  labels:
+    app.kubernetes.io/name: psp
```
Comment (vdemeester) [inline config/101-podsecuritypolicy.yaml:20]: Any reason for `psp` instead of a longer name ?

Comment (eddycharly) [reply config/101-podsecuritypolicy.yaml:20]: Nope, no particular reason, do you prefer a longer name ?
I don't think anyone will need to look this up anyway, i have set a name on every resource for consistency though.
Not completely sure if making the `app.kubernetes.io/name` mandatory for every resource makes sense to be honest.

**Comment by eddycharly** [comment]
/hold
I feel like there are probably some superfluous labels

File: config/101-podsecuritypolicy.yaml (hunk 1 above)
Comment (eddycharly) [reply config/101-podsecuritypolicy.yaml]: I removed the label where it was not necessary

**Comment by eddycharly** [comment]
/hold cancel
I removed `app.kubernetes.io/name` and `app.kubernetes.io/version` when they had no value.

@vdemeester how about using `app.kubernetes.io/version` everywhere instead of `pipeline.tekton.dev/release`, `triggers.tekton.dev/release`, `dashboard.tekton.dev/release`, etc ?

**Comment by vdemeester** [comment]
> /hold cancel
> I removed `app.kubernetes.io/name` and `app.kubernetes.io/version` when they had no value.
> 
> @vdemeester how about using `app.kubernetes.io/version` everywhere instead of `pipeline.tekton.dev/release`, `triggers.tekton.dev/release`, `dashboard.tekton.dev/release`, etc ?

yeah we could, but we will need to keep those (at least for a while) there as tools depend on them.

**Comment by eddycharly** [comment]
I know the dashboard depends on it, i am adding support for `app.kubernetes.io/veersion` there.
Any other tool ?

**Comment by vdemeester** [comment]
> I know the dashboard depends on it, i am adding support for `app.kubernetes.io/veersion` there.
> Any other tool ?

- `tkn`
- OpenShift Pipeline
- most likely a bunch of other tools that we do not control :stuck_out_tongue_closed_eyes: 

We should add the new one, mark the old one as "deprecated" (in release note) and decide on a time where it gets removed (if we want to remove it :stuck_out_tongue: )

**Comment by eddycharly** [comment]
@vdemeester do you want to include changes from https://github.com/tektoncd/pipeline/pull/2508 in this PR ?

**Comment by vdemeester** [comment]
> @vdemeester do you want to include changes from #2508 in this PR ?

oh yeah we can :+1:

**Comment by eddycharly** [comment]
Done, i took the `objectSelector` approach, hope it's ok for you.

**Pull request merged**